core/space/services/services_sharing.go

```diff
@@ -222,3 +224,39 @@ func (s *Space) OpenSharedFile(ctx context.Context, hash, password, filename str
 		Location: decryptedFile.Name(),
 	}, nil
 }
+
+func (s *Space) ShareFilesViaPublicKey(ctx context.Context, bucketName string, paths []string, pubkeys []crypto.PubKey) error {
```

Comment:
Just out of curiosity, how fast is this function overall? Especially, if we have quite a number `pubkeys` we want to share the files with.

Reply:
That's a good point, I guess its going to be as slow as much as there are pubkeys.  Small update on this though I think we can update the `SetMembers` interface to take the collection of pubkeys so we can just have one call there.  And re: the messaging that is temporary since it will be within the bucket ACL api textile is adding but think we can wait to see performance once the actual implementation is done.